src/js/DropContainer.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import classNames from 'classnames';
+import accepts from 'attr-accept';
+import css from '../stylesheets/dropcontainer.css';
+import UploadButton from './UploadButton';
+import ImagePreview from './ImagePreview';
+
+export default class DropContainer extends React.Component {
+   /**
+      * Base Container
+      * @param  {Object} props
+   */
+   constructor (props) {
+      super(props);
+
+      this.state = {
+         fileSrc: '',
+         dropzoneStyle: '',
+         isDragActive: false,
+         visibility: 'show'
+      };
+      this.onDragOver = this.onDragOver.bind(this);
+      this.onDrop = this.onDrop.bind(this);
+      this.onDragLeave = this.onDragLeave.bind(this);
+      this.onDragEnter = this.onDragEnter.bind(this);
+      this.handleChange = this.handleChange.bind(this);
+      this.handleInputButtonClick = this.handleInputButtonClick.bind(this);
+   }
+
+   componentDidMount () {
+      this.enterCounter = 0;
+   }
+
+   /**
+      * Handle the drag enter event
+      * @param  {Event]} event
+   */
+   onDragEnter (event) {
+      event.preventDefault();
+
+      // Count the dropzone and any children that are entered.
+      ++this.enterCounter;
+
+      // This is tricky. During the drag even the dataTransfer.
+      // files is null But Chrome implements some drag store,
+      // which is accesible via dataTransfer.items
+      const dataTransferItems = this.extractData(event),
+
+      // Now we need to convert the DataTransferList to Array
+      fileFormat = this.fileFormatFinder(this.props.accept),
+
+      // Check to make sure file is valid based on
+      // user's file format input
+      isFileValid = this.validateFileFormat(
+         Array.prototype.slice.call(dataTransferItems), fileFormat);
+
+      this.setState({
+         isDragActive: isFileValid,
+         dropzoneStyle: 'dragBorder',
+         visibility: 'hide'
+      });
+
+      // DropContainer turns green if valid format
+      // Otherwise turns red for warning
+      if (isFileValid) {
+         this.correctFormatBorderStyle()
+      } else {
+         this.incorrectFormatBorderStyle();
+      }
+   }
+
+   /**
+      * Handle the drag over event
+      * @param  {Event]} event
+   */
+   onDragOver (event) {
+      event.preventDefault();
+      event.stopPropagation();
+      event.dataTransfer.dropEffect = 'copy';
+      return false;
+   }
+
+   /**
+      * Handle the drag leave event
+   */
+   onDragLeave (event) {
+      event.preventDefault();
+
+      // Only deactivate once the dropzone and all children was left.
+      if (--this.enterCounter > 0) {
+         return;
+      }
+
+      this.setState({
+         isDragActive: false,
+         isDragReject: false,
+         dropzoneStyle: '',
+         visibility: 'show'
+      });
+
+      // if (this.props.onDragLeave) {
```

Comment:
We should remove commented code if it's not being used. It's not clear what's going on here.